crates/iceberg-rest-server/src/api/v1/warehouse.rs

```diff
@@ -75,6 +105,113 @@ pub trait WarehouseService<C: Catalog, A: AuthZHandler, S: SecretStore> {
             warehouse_id: warehouse_id.into_uuid(),
         })
     }
+
+    async fn update_warehouse(
+        request: UpdateWarehouseRequest,
+        context: ApiContext<State<A, C, S>>,
+        _headers: HeaderMap,
+    ) -> Result<UpdateWarehouseResponse> {
```

Comment:
Here, as well as in the create_warehouse request there is no check for AuthZ yet (using the A generic).
Could you add those as well?